application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,165 @@
+        OptionData anonymousOption = new OptionData(OptionType.BOOLEAN, OPTION_STAY_ANONYMOUS,
+                "if set, your name is hidden", true);
+
+        List<Command.Choice> choices = jda.getGuildCache()
+            .stream()
+            .map(guild -> new Command.Choice(guild.getName(), guild.getIdLong()))
+            .toList();
+
+        guildOption.addChoices(choices);
+
+        getData().addOptions(messageOption, guildOption, anonymousOption);
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        long userId = event.getUser().getIdLong();
+        event.deferReply().setEphemeral(true).queue();
+
+        if (isChannelOnCooldown(userId)) {
+            event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+                .setEphemeral(true)
+                .queue();
+            return;
+        }
+        authorIdToLastCommandInvocation.put(userId, Instant.now());
+
+        Optional<TextChannel> modMailAuditLog = getChannel(event);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn("Cannot find the designated modmail channel in this server.");
+            return;
+        }
+
+        String userMessage = event.getOption(OPTION_MESSAGE).getAsString();
+        boolean wantsToStayAnonymous = event.getOption(OPTION_STAY_ANONYMOUS).getAsBoolean();
+        String user = event.getUser().getAsMention();
+        MessageCreateAction message = modMailAuditLog.orElseThrow()
+            .sendMessageEmbeds(createModMailMessage(user, userMessage))
+            .addActionRow(DiscordClientAction.General.USER.asLinkButton("Click to see profile!",
+                    String.valueOf(userId)));
+
+        if (wantsToStayAnonymous) {
+            user = "Anonymous";
+            message = modMailAuditLog.orElseThrow()
+                .sendMessageEmbeds(createModMailMessage(user, userMessage));
+        }
+
+        message.mapToResult().flatMap(result -> {
+            if (result.isSuccess()) {
+                return event.getHook().editOriginal("Your message has been forwarded, thanks.");
+            } else {
+                return event.getHook()
+                    .editOriginal("There was an issue forwarding your message, sorry. We are "
+                            + "investigating.");
+            }
+        }).queue();
+
+    }
+
+    private MessageEmbed createModMailMessage(String user, String userMessage) {
+        return new EmbedBuilder().setDescription("**/modmail from %s** ".formatted(user))
+            .setFooter(userMessage)
```

Comment:
i might have missed a few discussions, but it seems super weird to me that u picked description+footer instead of title+description. why?

Reply:
also, i would prefer having the user as `.setAuthor(...)` (or was it called setUser?) in the embed. Would look better, UX wise.

So maybe the title could be "Modmail", the description the users message and the author (with profile picture) is the user who send it - or if anonymous just the name Anonymous.

u then also dont need that "profile"-button anymore, since discord offers this itself already on embed authors (i believe)

Reply:
Something like we have with `whois`?

![image](https://user-images.githubusercontent.com/30927961/194724670-d3c05e12-c619-4067-92e3-80ee551e722a.png)


```java
.setAuthor(member.getEffectiveName(), member.getEffectiveAvatarUrl(),
                    member.getEffectiveAvatarUrl())
```

Or `ScamBlocker`

```java
.setAuthor(author.getAsTag(), null, author.getAvatarUrl())```

I just tested these, they look nice, but they are not functional. I would really like to have a mention (or this button), so I can right-click and DM, or right-click copy ID. With this `setAuthor()` you don't have a right click, you get right click of the message, not the user.

It's the superior option to either just tag, just ID, or tag and profile picture is. You can see right away who actually sent the modmail, it's a human-readable name. And you can also access all the things you might need with right click. Best of both worlds.

That's why `whois` has `mention`, and `ScamBlocker` has an actionable quarantine button.

Reply:
We had a mention before, but tijs rightfully pointed out how it can crash. https://github.com/Together-Java/TJ-Bot/pull/493#discussion_r988576242

I don't really care how it's solved, it would just like for it to be clickable in any way, so I don't have to search the user or use another command. ^^

Reply:
good point. but adding the avatar and the name to that place costs us nothing, given that we have the user at hand. so instead of writing `setTitle("modmail from " + user.getName())` we would do `.setAuthor(user.getAsTag(), null, author.getAvatarUrl())` and `setTitle("Modmail")`.

But yeah, we have to keep the button in that case, thats fair.

Reply:
So we are keeping the button, but we are changing the` setTitle("modmail from " + user.getName())` to `.setAuthor(user.getAsTag(), null, author.getAvatarUrl()) .... setTitle("Modmail")`

Reply:
@Zabuzard

Reply:
yes

Reply:
this wasnt done yet

Reply:
```java
.setTitle("Modmail")
.setAuthor(user.getAsTag(), null, author.getAvatarUrl())
.setDescription(userMessage)
```